### PR TITLE
Signature and reference generation tweaks

### PR DESCRIPTION
**Pull request by bonnici**
Follow-up from https://github.com/apollographql/router/pull/4796 which addresses some performance concerns and non-blocking issues from the PR.

Related to https://apollographql.atlassian.net/browse/FED-117.

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Comment by bonnici** [comment]
@Geal these are some tweaks based on comments that you brought up in the original PR https://github.com/apollographql/router/pull/4796. I'm still not sure why the test isn't being run after I moved it to a separate file so any leads you can give me there would be appreciated.

**Review by bnjjj — APPROVED** [review]

**Review by garypen — APPROVED** [review]

File: apollo-router/src/apollo_studio_interop/test.rs [hunk 1]
```diff
@@ -0,0 +1,1419 @@
+#[cfg(test)]
+mod tests {
```
Comment (garypen) [inline apollo-router/src/apollo_studio_interop/test.rs:2]: Delete these two lines and the matching module close at the end. They are already in module `tests` by virtue of the file name.

File: apollo-router/src/apollo_studio_interop/mod.rs [hunk 2]
```diff
@@ -523,1423 +523,3 @@ fn format_value(value: &Value, f: &mut fmt::Formatter) -> fmt::Result {
         rest => f.write_str(&rest.to_string()),
     }
 }
-
-#[cfg(test)]
-mod tests {
```
Comment (garypen) [inline apollo-router/src/apollo_studio_interop/mod.rs:528]: Retain these two lines to include the `tests` module for testing.
e.g.:
```
#[cfg(test)]
mod tests;
```

Comment (garypen) [reply apollo-router/src/apollo_studio_interop/mod.rs:528]: also, you need to rename `test.rs` to `tests.rs`.

**Review by garypen — COMMENTED** [review]
To answer your questions about "why isn't the test working".

You need to make a couple of tweaks to your PR to comply with rust's module expectations.

File: apollo-router/src/apollo_studio_interop/mod.rs (hunk 2 above)
Comment (bonnici) [reply apollo-router/src/apollo_studio_interop/mod.rs:528]: Thank you! I tried some of these but not at the same time and I also was always missing the #[cfg(test)]; mod tests;` part.

**Review by Geal — APPROVED** [review]

**Review by garypen — APPROVED** [review]

**Review by o0Ignition0o — APPROVED** [review]

**Pull request merged**